arch/arm/core/Kconfig

```diff
@@ -166,11 +166,15 @@ config RUNTIME_NMI
 	  needed, enable this option and attach it via z_arm_nmi_set_handler().
 
 config PLATFORM_SPECIFIC_INIT
-	bool "Platform (SOC) specific startup hook"
+	bool "Platform (SOC) specific startup hook [DEPRECATED]"
+	default y if SOC_RESET_HOOK
+	select DEPRECATED
 	help
 	  The platform specific initialization code (z_arm_platform_init) is
 	  executed at the beginning of the startup code (__start).
```

Comment:
This comment is now very misleading. `z_arm_platform_init` isn't ever called anymore.  I think this is a bug.